### PR TITLE
`move`-Command Tests, Refactor, Fix

### LINKED ISSUE #11783: `move` to the same column creates duplicate keys on records
### Describe the bug

Using `move` on a record and moving a key to before/after itself will duplicate that key.

### How to reproduce

```Nushell
~> {a: 1} | move a --after a
╭───┬───╮
│ a │ 1 │
│ a │ 1 │
╰───┴───╯
```

### Expected behavior

An error.

### Screenshots

_No response_

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.89.0                                                       |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13) (Alpine Linux 1.72.1-r0) |
| cargo_version      | cargo 1.72.1                                                 |
| build_time         | 2024-01-13 03:02:49 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash, which, zip                           |
| installed_plugins  |                                                              |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dannou812**
fixes #11783 

# Description
Firstly Tests for the `move` Command have been added. Afterwards some duplicate Code has been removed and finally an error-message has been added for when a column is tried to be moved based on itself. This should fix #11783 .

To reiterate, the example of the initial issue now plays out as follows:
```shell
> {a: 1} | move a --after a                             
Error: nu::shell::incompatible_parameters

  × Incompatible parameters.
   ╭─[entry #1:1:15]
 1 │ {a: 1} | move a --after a
   ·               ┬         ┬
   ·               │         ╰── relative to itself
   ·               ╰── Column cannot be moved
   ╰────
``` 

# User-Facing Changes
The error message shown above.

# Tests + Formatting
I added some Tests for the behavior of the command. If I should add more, please let me know but I added everything that came to mind when thinking about the command.

### CODE REVIEW

**Comment by dannou812** [comment]
I am sorry for all the commits but somehow I am not getting all the clippy warnings on my local machine. If anyone knows what the problem could be please let me know (I am running `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used`)

File: crates/nu-command/src/filters/move_.rs [hunk 1]
```diff
@@ -284,10 +271,254 @@ fn move_record_columns(
...
+        let test_record = get_test_record(vec!["a", "b", "c", "d", "e"], vec![1, 2, 3, 4, 5]);
+        let after: Spanned<BeforeOrAfter> = Spanned {
+            item: BeforeOrAfter::After("c".to_string()),
+            span: test_span,
+        };
+        let columns = [
+            Value::String {
+                val: "b".to_string(),
+                internal_span: test_span,
+            },
```
Comment (fdncred) [inline crates/nu-command/src/filters/move_.rs:338]: I think all these `Value::String {}` instances should probably be `Value::test_string()`

**Comment by fdncred** [comment]
Other than my comment above I think this looks good. Thanks!

**Comment by dannou812** [comment]
Great, I will change and commit it immediately.

File: crates/nu-command/src/filters/move_.rs [hunk 2]
```diff
@@ -284,10 +271,200 @@ fn move_record_columns(
...
     use super::*;
 
+    // helper
+    fn get_test_record(columns: Vec<&str>, values: Vec<i64>) -> Record {
+        let test_span = Span::test_data();
+        Record::from_raw_cols_vals(
+            columns.iter().map(|col| col.to_string()).collect(),
+            values
+                .iter()
+                .map(|val| Value::Int {
```
Comment (fdncred) [inline crates/nu-command/src/filters/move_.rs:281]: Same thing for this too. I believe there's a `Value::test_int()`

**Comment by fdncred** [comment]
Thanks for helping make nushell better!

**Pull request merged**